### PR TITLE
[CosmosDB] GA PowerShell commands for InAccount Restore of Deleted Database/ Containers feature for SQL/Mongo/Table/Gremlin

### PR DESCRIPTION
**Pull request by ddhamotharan**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by dolauli** [comment]
@ddhamotharan Please fix the CI failures.

**Comment by ddhamotharan** [comment]
@dolauli Fixed all CI failures. Please review when you get a chance. thank you

**Review by dolauli — APPROVED** [review]
LGTM

File: src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+                    databaseAccount = lastestAccountToRestore;
+                }
+                else
+                {
+                    this.WriteWarning($"No database accounts found with matching account name {this.AccountName} that was alive");
+                    return;
+                }
+
+                Regex regex = new Regex(@"[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}");
```
Comment (amisi01) [inline src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs:125]: DatabaseAccountInstanceId?

File: src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                    {
+                        DateTime eventDateTime = DateTime.Parse(restorableGremlinDatabase.Resource.EventTimestamp);
+                        if (restorableGremlinDatabase.Resource.OperationType.Equals(OperationType.Delete) && latestDeleteTime < eventDateTime)
+                        {
+                            latestDeleteTime = eventDateTime;
+                        }
+                    }
+                }
+
+                utcRestoreDateTime = latestDeleteTime.AddSeconds(-2);
```
Comment (amisi01) [inline src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs:148]: -1?

File: src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs (hunk 1 above)
Comment (amisi01) [reply src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs:125]: similar to other files as well?

File: src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs (hunk 2 above)
Comment (amisi01) [reply src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinDatabase.cs:148]: similar in other files as well?

File: src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinGraph.cs [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                {
+                    latestDatabaseDeleteTime = DateTime.MaxValue;
+                }
+
+                IEnumerable restorableGremlinGraphs = CosmosDBManagementClient.RestorableGremlinGraphs.ListWithHttpMessagesAsync(
+                    databaseAccount.Location,
+                    accountInstanceId,
+                    databaseRid,
+                    latestDatabaseCreateOrRecreateTime.ToString(),
+                    (latestDatabaseCreateOrRecreateTime < latestDatabaseDeleteTime) ? latestDatabaseDeleteTime.ToString() : DateTime.MaxValue.ToString()).GetAwaiter().GetResult().Body;
```
Comment (amisi01) [inline src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinGraph.cs:176]: DateTime.MaxValue as restore time won't work, right?
can we simplify logic as following:
for snapshot : snapshots.sorted():
    if snapshot.type == "Delete": latestDeleteSnapshot = snapshot
    else if snapshot.type == "(Re)Create": latestDeleteSnapshot = null
    
 latestDeleteSnapshot == null : not restorable

**Pull request merged**